src/main/java/tc/oc/pgm/api/setting/SettingKey.java

```diff
@@ -18,10 +19,12 @@
   DEATH(
       Arrays.asList("death", "dms"), DEATH_ALL, DEATH_OWN), // Changes which death messages are seen
   PICKER("picker", PICKER_AUTO, PICKER_ON, PICKER_OFF), // Changes when the picker is displayed
-  OBSERVERS(
-      Arrays.asList("observers", "obs"),
-      OBSERVERS_ON,
-      OBSERVERS_OFF); // Changes if observers are visible
+  OBSERVERS(Arrays.asList("observers", "obs"), OBSERVERS_ON, OBSERVERS_OFF) {
+    @Override
+    public void update(MatchPlayer player) {
+      player.resetVisibility();
+    }
+  }; // Changes if observers are visible
```

Comment:
A way to heavily simplify this, is to have a constructor overload:
`SettingKey(List<String> aliases, Consumer<MatchPlayer> update, SettingValue... values)`

Then this just becomes:
`OBSERVERS(Arrays.asList("observers", "obs"), MatchPlayer::resetVisibility, OBSERVERS_ON, OBSERVERS_OFF)`

As obvious, you'll need to keep the consumer on a variable, and on the update method call it if it's not null

Reply:
The current version is imo simpler, and I don't think there are going to be many settings with custom update logic.

Reply:
For now, either way works